src/Spago/Command/Fetch.purs

```diff
@@ -357,31 +357,67 @@ writeNewLockfile reason allTransitiveDeps = do
 toAllDependencies :: PackageTransitiveDeps -> PackageMap
 toAllDependencies = foldl (Map.unionWith (\l _ -> l)) Map.empty
 
+data CopyOrMove = Copy | Move
+
 getGitPackageInLocalCache :: forall a. PackageName -> GitPackage -> Spago (Git.GitEnv a) Unit
 getGitPackageInLocalCache name package = do
-  let localPackageLocation = Config.getPackageLocation name (GitPackage package)
-  tempDir <- mkTemp' (Just $ printJson Config.gitPackageCodec package)
-  logDebug $ "Cloning repo in " <> tempDir
-  Git.fetchRepo package tempDir >>= case _ of
-    Left err -> die err
-    Right _ -> do
+  let gitCacheLocation = Config.getGlobalGitCacheForRepo package
+      localPackageLocation = Config.getPackageLocation name (GitPackage package)
+  existsGitCache <- FS.exists gitCacheLocation
+  Tuple copyOrMove sourceDir <-
```

Comment:
Don't think we need this `CopyOrMove` type - if the package can be cached globally then we need to copy it, if it can't then we need to move it.
Moreover, it's generally more idiomatic to use a Record rather than an N-tuple when returning multiple values:
```suggestion
  { globallyCached, sourceDir } <-
```

(where `globallyCached` is a boolean)